Add WAL content for 2.0 release

Andriciuc opened the pull request:
- remove (tech preview)
- remove mentions of WAL being BETA and warning notes

Andriciuc committed:
Add WAL content for 2.0 release

Andriciuc committed:
improve functions.md with small text fix and reword introductory paragraph for Global Principal key config

- fix a small space in functions and add * to pg_tde_set_
- reworded introductory paragraph in Global Principal Key configuration, added descriptions for the two functions used to config the global principal key
- added explanatory paragraph to Next steps for clarity

Andriciuc committed:
Merge branch 'TDE_REL_17_STABLE' into wal-2.0-update

Andriciuc committed:
Merge branch 'TDE_REL_17_STABLE' into wal-2.0-update

Andriciuc committed:
Merge branch 'TDE_REL_17_STABLE' into wal-2.0-update

Andriciuc committed:
add WAL tool support to limitations, improve flow, add button to setup

nastena1606 commented on contrib/pg_tde/documentation/docs/index/tde-limitations.md:
```diff
@@ -1,7 +1,29 @@
+    * `pg_rewind`
+    * `pg_upgrade`
+    * `pg_waldump`
+    * pgBackRest
```
why is this not code-indented?

Andriciuc replied:
fixed

nastena1606 commented on contrib/pg_tde/documentation/docs/index/tde-limitations.md:
```diff
@@ -1,7 +1,29 @@
 
 [View the versions and supported deployments :material-arrow-right:](supported-versions.md){.md-button}
+
+Begin the installation process when you’re ready to set up encryption.
```
```suggestion
Begin the installation process when you're ready to set up encryption.
```

nastena1606 commented on contrib/pg_tde/documentation/docs/index/tde-limitations.md:
```diff
@@ -1,7 +1,29 @@
 # Limitations of pg_tde
 
-The following are current limitations of `pg_tde`:
+The following limitations apply to the current release:
```
IMHO, it would be useful to specify the release version here instead of the common "current"

Andriciuc replied:
oh yeah! thank you

Andriciuc committed:
Update contrib/pg_tde/documentation/docs/index/tde-limitations.md

Co-authored-by: Anastasia Alexandrova <anastasia.alexandrova@percona.com>

Andriciuc committed:
update limitations with feedback

Andriciuc committed:
Merge branch 'wal-2.0-update' of https://github.com/percona/postgres into wal-2.0-update

Andriciuc committed:
Merge branch 'TDE_REL_17_STABLE' into wal-2.0-update

Andriciuc committed:
add limitation regarding WAL shipping standy not supported with WAL encryption

Andriciuc committed:
remove confusing text from how tde works, link in note to KMS config for user

* add  table for user to select preferred KMS config

Andriciuc committed:
add mention of open source and enterprise ed being supported for pg_tde

Andriciuc committed:
Update the Features topic buttons for better clarity (#508)

Andriciuc committed:
Merge branch 'TDE_REL_17_STABLE' into wal-2.0-update

jeltz commented on contrib/pg_tde/documentation/docs/global-key-provider-configuration/overview.md:
```diff
@@ -9,10 +9,15 @@ To use an external KMS with `pg_tde`, follow these two steps:
+| **Fortanix**       | Fortanix DSM key management.                          | [Configure Fortanix →](kmip-fortanix.md) |
+| **Thales**         | Thales CipherTrust Manager and DSM.                   | [Configure Thales →](kmip-thales.md) |
+| **OpenBao**        | Community fork of Vault, supporting KV v2.            | [Configure OpenBao →](kmip-openbao.md) |
+| **Keyring file** *(not recommended)* | Local key file for dev/test only.                  | [Configure keyring file →](keyring.md) |
```
This table feels a bit weird, or at least in combiantion with "Select your preferred configuration from the following table:".

Andriciuc replied:
I see what you mean yeah, what about changing that paragraph to: 

`pg_tde` supports the following providers:

[table]

I wanted to somehow present them on the same page (keeping their eyes in line so to speak) and had tested with cards, tables, links... this table I think seems like a compromise between whatever future providers we will add, and a simple stylish easy to grasp presentation. 🤔

jeltz replied:
Yeah, makes sense but if we change to "spports" we need to explain that it is only tested with some KMIP providers and may not work woth all.

Andriciuc replied:
Good point! I'll leave it as: 

`pg_tde` has been tested and approved with the following key providers:

[table]

jeltz commented on contrib/pg_tde/documentation/docs/index/tde-limitations.md:
```diff
@@ -1,8 +1,30 @@
 # Limitations of pg_tde
 
-The following are current limitations of `pg_tde`:
+Limitations of `pg_tde` (applies to release {{release}}):
```
Doesn't all of the documentation only apply to this release? Feels like a bit weird thing to point out here specifcally, but maybe the idea is to be very clear to people who arrive in the future to the wrong version.

Andriciuc replied:
Good catch yeah! I thought of adding the note because the limitations in 1.0 and 2.0 differ, and we want to make sure users don’t confuse them. Since the 1.0 docs will remain archived in Git but often still surface in searches, I thought maybe the explicit version reference helps users quickly confirm they’re looking at the right release if that makes sense.

jeltz replied:
Maybe  make it short by just writing "Limitations of pg_tde 2.0".

Andriciuc replied:
Done!

jeltz commented on contrib/pg_tde/documentation/docs/index/tde-limitations.md:
```diff
@@ -1,8 +1,30 @@
-* The WAL encryption feature is currently in beta and is not effective unless explicitly enabled. It is not yet production ready. **Do not enable this feature in production environments**.
+* PostgreSQL’s internal system tables, which include statistics and metadata, are not encrypted.
+* `pg_createsubscriber` is not supported.
+* WAL shipping standby (warm standby) is not supported with WAL encryption. You can instead use [Streaming Replication with `tde_heap`](../replication.md), which is supported with encrypted WAL.
```
This is not true. What is not supported is something much more specific which barely deserves a mention but maybe does. Let's keep that as a separate PR.

It should almost be on the architecture page if anywhere. Or maybe on the page which documents our archive and restore helpers.

Andriciuc replied:
Hmm I see, thank you for clarifying that! I removed the WAL shipping mention.

jeltz commented on contrib/pg_tde/documentation/docs/index/tde-limitations.md:
```diff
@@ -1,8 +1,30 @@
+* PostgreSQL’s internal system tables, which include statistics and metadata, are not encrypted.
+* `pg_createsubscriber` is not supported.
+* WAL shipping standby (warm standby) is not supported with WAL encryption. You can instead use [Streaming Replication with `tde_heap`](../replication.md), which is supported with encrypted WAL.
+
```
Shouldn't we mention that temporary files are not encrypted?

Andriciuc replied:
Just to confirm we're talking about `work_mem` spillover or something else? If that's the case I propose adding this:

* Temporary files created when queries exceed `work_mem` are not encrypted. These files may persist if the query runs for a long time or the server crashes which can expose sensitive data in plaintext on disk.

We need to be specific when asking which temp files are not encrypted. 🤔

Andriciuc replied:
Added!

jeltz commented on contrib/pg_tde/documentation/docs/index/tde-limitations.md:
```diff
@@ -1,8 +1,30 @@
+
+??? note "Click to expand"
+    * Patroni
+    * `pg_basebackup` (with `--wal-method=stream`)
```
Also with `--wal-method=none` but maybe too obvious?

Andriciuc replied:
never a bad thing to add more info! Saves the user a question they would have for our support, but since we're talking about WAL encryption specifically, would `--wal-method=none` be useful here?

jeltz replied:
Yes! You use `--wal-method=none` if you want to build your own pgBackRest. The idea is that you set `archive_command` to somehting and start archiving WAL and then run `pg_basebackup` with `--wal-method=none` that way you can restore the backup by combining it with the archived WAL files.

Andriciuc replied:
Great explanation, thank you, now I get it!

I added the none method and a link to our basebackup topic if the user wants more info:

* `pg_basebackup` (with `--wal-method=stream` or `--wal-method=none`), for details on using `pg_basebackup` with WAL encryption, see [Backup with WAL encryption enabled](../how-to/backup-wal-enabled.md)

Note: the link doesn't 'work' because we didn't merge yet but it's the correct path.

jeltz commented on contrib/pg_tde/documentation/docs/index/tde-limitations.md:
```diff
@@ -1,8 +1,30 @@
+The following tools and extensions in Percona Distribution for PostgreSQL have been tested and verified to work with `pg_tde` WAL encryption:
+
+??? note "Click to expand"
+    * Patroni
```
Maybe we should explain which configs because we certainly do not support all.

Andriciuc replied:
added an ## Example Patroni configuration subtopic and linked to Patroni mention in the WAL compatibility list

jeltz commented on contrib/pg_tde/documentation/docs/index/tde-limitations.md:
```diff
@@ -1,8 +1,30 @@
+    * `pg_rewind`
+    * `pg_upgrade`
+    * `pg_waldump`
+    * `pgBackRest`
```
The tool should probably not be in code format since the command line tool is called `pgbackrest`.

Andriciuc replied:
Fixed!

jeltz commented on contrib/pg_tde/documentation/docs/index/tde-limitations.md:
```diff
@@ -1,8 +1,30 @@
+
+The following tools and extensions in Percona Distribution for PostgreSQL have been tested and verified to work with `pg_tde` WAL encryption:
+
+??? note "Click to expand"
```
Technically a lot more tools have been tested but I assume these are the ones which touch the WAL?

Andriciuc replied:
Yep! Specifically from the Tooling support for encrypted WAL spreadsheet. As a new user I'd appreciate having this clear and in the open, hope it helps others too!

jeltz commented on contrib/pg_tde/documentation/docs/wal-encryption.md:
```diff
@@ -118,3 +115,5 @@ Now WAL files start to be encrypted for both encrypted and unencrypted tables.
 For more technical references related to architecture, variables or functions, see:
 [Technical Reference](advanced-topics/tech-reference.md){.md-button}
+
+💬 Need help customizing this for your infrastructure? [Contact Percona support :octicons-link-external-16:](get-help.md)
```
Do we typically use uncode like this?

Andriciuc replied:
I added the 💬 to make the support note stand out a bit, I want to test and see if small, consistent icons that make key content ‘pop’ can potentially improve user retention 🤔 

Edit: Other products use varying degrees of this method and I would like to test it out for ours as well

Andriciuc committed:
modify limitations and global key config with feedback

Andriciuc committed:
add none method to basebackup and link to topic

Andriciuc committed:
rename WAL tool support to compatibility

Disliked how in Limitations we put tool support so I rewrote the title to better reflect the limitation chapter scope

Andriciuc committed:
add Example Patroni configuration for Patroni tool

This ensures that the user has a configuration example for Patroni, at least offered by us.

jeltz commented on contrib/pg_tde/documentation/docs/global-key-provider-configuration/overview.md:
```diff
@@ -9,10 +9,15 @@ To use an external KMS with `pg_tde`, follow these two steps:
+     While key files may be acceptable for **local** or **testing environments**, KMS integration is the recommended approach for production deployments.
 
-Select your prefered configuration from the links below:
+`pg_tde` has been tested and approved with the following key providers:
```
I think we could remove approved.

Andriciuc replied:
removed!

Andriciuc replied:
Removed 👍

jeltz reviewed (approved):
I want to change some things but I prefer this to be merged now.

Andriciuc committed:
add a new paragraph for unsupported tools under WAL tool compat and remove from limitations

Andriciuc committed:
improve supported vs unsupported tools section

- removed approved from overview for key providers

Andriciuc committed:
Merge branch 'release-17.5.3' into wal-2.0-update

Andriciuc committed:
Merge branch 'release-17.5.3' into wal-2.0-update
